### ISSUE TITLE
Field location is not stored in dump files

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#736]
That would be nice, so that the `boutcore.Field3D.fromCollect()` could return a Field3D with an appropriate location.

Maybe an additional char/string variable could be added, to store for each field the location, something like `<name>_location`?

**Comment by johnomotani** [comment]
In both NetCDF and HDF5 files you can add attributes to variables.
http://www.unidata.ucar.edu/software/netcdf/docs/group__attributes.html
https://support.hdfgroup.org/HDF5/doc1.6/UG/13_Attributes.html

This might be neater than adding extra variables, but requires adding code to each fileio implementation.

**Comment by bendudson** [comment]
Agree, this would be very nice. Attributes sounds like a good way to go, though string support is also on the list of requests (issues #222 and #426 ). 
The output system really needs changing, so that it doesn't depend on storing pointers to user variables. This would be a breaking change. For now we're stuck with the original design, so I may find time to implement this.
